rasa/cli/test.py

```diff
@@ -131,11 +135,15 @@ def test_nlu(args: argparse.Namespace) -> None:
                 )
                 continue
 
-        output = args.report or DEFAULT_NLU_RESULTS_PATH
+        report_output = args.report or DEFAULT_NLU_RESULTS_PATH
+
+        if output:
+            report_output = os.path.join(output, report_output)
```

Comment:
I guess, we don't need that. `--report` is used for `Output path to save the intent/entity metrics report.`. So, if we do compare testing, I would store the results just in the `--out` folder. No need to "misuse" the report folder here anymore.